components/core/checkbox/Checkbox.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+    <label
+        class="
+            checkbox
+            mt-3
+            ml-2
+            py-2
+            px-3
+            border
+            rounded-lg
+            transition-all
+            cursor-pointer
+            text-xs
+            lg:text-base
+        "
+        :class="{ checked: isChecked }"
+    >
+        <input
+            :id="id"
+            type="checkbox"
+            :value="id"
+            :checked="isChecked"
+            class="hidden"
+            @change="toggleCheckbox()"
+        />
+        <fa
+            :icon="isChecked ? ['far', 'check-square'] : ['far', 'square']"
+            aria-hidden="true"
+            class="mr-2 lg:mr-6"
+        ></fa>
+        {{ label }}
+    </label>
+</template>
+
+<script>
+export default {
+    name: 'CoreCheckbox',
+    props: {
+        id: {
+            type: String,
+            default: '',
+        },
+        label: {
+            type: String,
+            default: '',
+        },
+        value: {
+            type: [Array, Boolean],
+            default() {
+                return []
+            },
+        },
+    },
+    computed: {
+        isChecked() {
+            if (typeof this.value === 'boolean') {
+                return this.value
+            } else {
+                return this.value.includes(this.id)
+            }
+        },
+    },
```

Comment:
How about directly passing `isChecked` props to this component? Defining whether a checkbox should be checked in the caller of the checkbox component would be better.

Reply:
Yes, would be much more readable.